### DISCUSSION TITLE
rust-course/cookbook 替换被弃用的部分

### DISCUSSION THREAD

**Discussion by LyntNy4n** [sunface/rust-course#701] [内容建议]
目前已知有两个地方
一个是参数解析[https://course.rs/cookbook/cmd/parsing.html](url)
clap在新版弃用了App 改用Command了
![}D_V49L({SK 640NOC2G{Q6](https://user-images.githubusercontent.com/49772385/161370333-b87a0c4e-f1b4-4d29-b28a-9d23dbaa6eb2.png)
另一个地方是错误处理 目前还没有翻译到该部分 不过原版是有的
其中的error_chain 库已被弃用[https://users.rust-lang.org/t/error-chain-is-no-longer-maintained/27561](url)
现在应该是用anyhow和thiserror吧

**Comment by sunface** [comment]
是的，这本书也是许久没更新了，如果可以，能否给 clap 提一个 pr，感觉你比较熟悉 :)

**Reply by LyntNy4n** [reply]
我是别人提到才顺便反映了一下 我自己还没用过clap 不太敢乱改 😥

**Reply by sunface** [reply]
好吧，我先记录了。。谢谢哈
